### PR TITLE
Fix: Object3D getObjectByProperty recursion

### PR DESCRIPTION
**Pull request by ghost**
Fix to Object3D.getObjectByProperty method which did not respect the
recursive flag. Implemented default recursive equals false state found
in documentation.

### CODE REVIEW

File: src/core/Object3D.js [hunk 1]
```diff
@@ -390,18 +390,24 @@ THREE.Object3D.prototype = {
...
 		if ( this[ name ] === value ) return this;
+		if ( recursive === undefined ) recursive = false;
 
 		for ( var i = 0, l = this.children.length; i < l; i ++ ) {
 
 			var child = this.children[ i ];
-			var object = child.getObjectByProperty( name, value, recursive );
 
-			if ( object !== undefined ) {
+			if ( recursive ) {
```
Comment (mrdoob) [inline src/core/Object3D.js]: I would you do `if ( recursive === true ) {`.

File: src/core/Object3D.js [hunk 2]
```diff
@@ -390,18 +390,24 @@ THREE.Object3D.prototype = {
 	getObjectByProperty: function ( name, value, recursive ) {
 
 		if ( this[ name ] === value ) return this;
+		if ( recursive === undefined ) recursive = false;
```
Comment (mrdoob) [inline src/core/Object3D.js]: I would remove this line.

**Comment by darryllehmann** [comment]
Sounds good, will do.

**Comment by mrdoob** [comment]
Hmmm... Did you just noticed that this was broken or did you needed it to not recurse?

**Comment by darryllehmann** [comment]
My use case didn't require it to not recurse, I simply noticed during testing it wasn't delivering the expected output.

**Comment by mrdoob** [comment]
Ok. I think I'll remove the `recurse` parameter then. I think it's a over-engineering case.
Thanks for making me notice :)

**Pull request closed without merging**